src/services/ExposureNotificationService/ExposureNotificationService.ts

```diff
@@ -341,6 +344,120 @@ export class ExposureNotificationService {
     }
   }
 
+  public getTotalSeconds = (scanInstances: ScanInstance[]) => {
+    const totalSeconds = scanInstances
+      .map(x => x.secondsSinceLastScan)
+      // with initial value to avoid when the array is empty
+      .reduce((partialSum, x) => partialSum + x, 0);
+    return totalSeconds;
+  };
+
+  public getAttenuationDurations = (scanInstances: ScanInstance[], attenuationDurationThresholds: number[]) => {
+    //  typicalAttenuation values are in positive DB units
+    const immediateScans = scanInstances.filter(x => x.typicalAttenuation <= attenuationDurationThresholds[0]);
+    const nearScans = scanInstances.filter(x => {
+      return (
+        x.typicalAttenuation > attenuationDurationThresholds[0] &&
+        x.typicalAttenuation <= attenuationDurationThresholds[1]
+      );
+    });
+    const farScans = scanInstances.filter(x => x.typicalAttenuation > attenuationDurationThresholds[1]);
+    return [this.getTotalSeconds(immediateScans), this.getTotalSeconds(nearScans), this.getTotalSeconds(farScans)];
+  };
+
+  public async checkIfExposedV2({
+    exposureWindows,
+    attenuationDurationThresholds,
+    minimumExposureDurationMinutes,
+  }: {
+    exposureWindows: ExposureWindow[];
+    attenuationDurationThresholds: number[];
+    minimumExposureDurationMinutes: number;
+  }): Promise<[boolean, ExposureSummary | undefined]> {
+    if (exposureWindows.length === 0) {
+      return [false, undefined];
+    }
+    const dailySummariesObj: {[key: string]: {attenuationDurations: number[]; matchedKeyCount: number}} = {};
+    exposureWindows
+      .filter(window => {
+        if (window.infectiousness === Infectiousness.None) {
```

Comment:
This `if` statement can be simplified, combined with the `return` statement.

Reply:
I find my version easier to read but I will go with your suggestion anyway 😉